lib/docker/template/cache.rb

```diff
@@ -42,9 +42,10 @@ def aliased_context(builder)
       # --
 
       def cleanup(repo)
+        return unless repo.clean_cache?
         cache_dir = repo.cache_dir.parent
 
-        if repo.cacheable? && cache_dir.exist?
+        if cache_dir.exist?
```

Comment:
It looks like you're fine looking at the repo.rb to see what cachable? does, and looking at my notes, it was slated to be renamed anyways, it could still have negative impact and if you wish to manually test that `envygeeks/docker` has a few rootfs images (ubuntu and alpine) that can be built to see how it impacts clean and build. `docker-template build alpine:3.2 --args`